SpeechResponder/CustomFunctions/BlueprintDetails.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Cottle.Functions;
+using Cottle.Values;
+using EddiDataDefinitions;
+using EddiSpeechResponder.Service;
+using JetBrains.Annotations;
+using System;
+
+namespace EddiSpeechResponder.CustomFunctions
+{
+    [UsedImplicitly]
+    public class BlueprintDetails : ICustomFunction
+    {
+        public string name => "BlueprintDetails";
+        public FunctionCategory Category => FunctionCategory.Details;
+        public string description => @"
+This function will provide full information for a blueprint, given its name and grade.
+
+BlueprintDetails() takes two mandatory arguments: the name of the blueprint and the grade to retrieve. 
+
+Common usage of this is to provide further information about a blueprint, for example:
+
+    {set blueprint to BlueprintDetails(""Dirty Drive Tuning"", 5)}
+    {len(blueprint.materials)} {if len(blueprint.materials) > 1: different materials are |else: material is} required to produce {blueprint.localizedName}.";
+        public NativeFunction function => new NativeFunction((values) =>
```

Comment:
Sorry, a thought just struck me. Isn't this allocating a new instance of the NativeFunction every time the property is accessed? Don't we instead want to say:

```cs
    public NativeFunction function = new NativeFunction((values) =>
```
Or is this some form of [lazy initialization](https://docs.microsoft.com/en-us/dotnet/framework/performance/lazy-initialization)?